tools/RAiDER/processWM.py

```diff
@@ -140,16 +136,33 @@ def prepareWeatherModel(
         del weather_model
 
 
-def checkBounds(weather_model, outLats, outLons):
-    '''Check the bounds of a weather model'''
-    ds = xr.load_dataset(weather_model.files[0])
-    coords = ds.coords  # coords is dict-like
-    keys = [k for k in coords.keys()]
-    xc = coords[keys[0]]
-    yc = coords[keys[1]]
-    lat_bounds = [yc.min(), yc.max()]
-    lon_bounds = [xc.min(), xc.max()]
-    self_extent = lat_bounds + lon_bounds
-    in_extent = weather_model._getExtent(outLats, outLons)
+def checkContainment(weather_model: weatherModel,
+                     outLats: np.ndarray,
+                     outLons: np.ndarray) -> bool:
+    '''Check if weather model contains bounding box of outLats and outLons'''
+    weather_model_path = weather_model.files[0]
+    with rasterio.open(f'netcdf:{weather_model_path}') as ds:
+        datasets = ds.subdatasets
+
+    with rasterio.open(datasets[0]) as ds:
+        bounds = ds.bounds
+
+    xmin, ymin, xmax, ymax = tuple(bounds)
+    weather_model_box = box(xmin, ymin, xmax, ymax)
+
+    xmin_input, xmax_input = np.min(outLons), np.max(outLons)
+    ymin_input, ymax_input = np.min(outLats), np.max(outLats)
+    input_box = box(xmin_input, ymin_input, xmax_input, ymax_input)
+
+    # Logger
+    input_box_str = [f'{x:1.2f}' for x in [xmin_input, ymin_input,
```

Comment:
I think we want to avoid logging inside subfunctions, as we don't necessarily want to log when running unit tests etc. You could move this part into the main "prepareWeatherModel" function.

Reply:
Could we remove the logging?

Reply:
We definitely want to log the bounds, as that is the first thing you look at when there is a problem with the extents. I originally had the function return the extents, rather than return the flag.